include/boost/beast/http/impl/basic_parser.ipp

```diff
@@ -112,26 +112,9 @@ loop:
 
     case state::start_line:
     {
-        maybe_need_more(p, n, ec);
+        parse_start_line(p, n, ec);
         if(ec)
             goto done;
-        parse_start_line(p, p + (std::min<std::size_t>)(
-            header_limit_, n), ec, is_request{});
-        if(ec)
-        {
-            if(ec == error::need_more)
-            {
-                if(n >= header_limit_)
-                {
-                    BOOST_BEAST_ASSIGN_EC(ec, error::header_limit);
-                    goto done;
-                }
-                if(p + 3 <= p1)
-                    skip_ = static_cast<
-                        std::size_t>(p1 - p - 3);
-            }
-            goto done;
-        }
         BOOST_ASSERT(! is_done());
         n = static_cast<std::size_t>(p1 - p);
         if(p >= p1)
```

Comment:
@vinniefalco I couldn't find any scenario that makes `p > p1` possible. Do you remember the rationale behind this line? If the only possibility is `p == p1`, we can completely remove this branch and allow `parse_fields` to be called with `n = 0`

Reply:
When I write these types of conditions, I make them inequalities on purpose, for no particular reason.